src/Server/HTTPHandler.h

```diff
@@ -69,6 +71,22 @@ class HTTPHandler : public HTTPRequestHandler
 
     CurrentMetrics::Increment metric_increment{CurrentMetrics::HTTPConnection};
 
+    // The request_context and the request_credentials instances may outlive a single request/response loop.
+    // This happens only when the authentication mechanism requires more than a single request/response exchange (e.g., SPNEGO).
+    std::unique_ptr<Context> request_context;
```

Comment:
`HTTPHandler` is normally recreated for each request, isn't it?

Reply:
I remember now, this is made future-proof, for cases when HTTP connection also means a single same session, i.e. authenticated only once. That would imply support for executing requests on the same handler instance. And then (independently) the multi-token SPNEGO could also become possible (they can happen theoretically, but Kerberos manages to fit everything in a single token just by good fortune).

Reply:
Currently, the Poco implementation of HTTP stuff recreates HTTPHandler instance, yes.